operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java

```diff
@@ -102,6 +104,13 @@ public UpdateControl<Keycloak> reconcile(Keycloak kc, Context<Keycloak> context)
 
         Log.infof("--- Reconciling Keycloak: %s in namespace: %s", kcName, namespace);
 
+        if (kc.getSpec().getInstances() == null) {
+            // explicitly set defaults - and let another reconciliation happen
+            // this avoids ensuring unintentional modifications have not been made to the cr
+            kc.getSpec().setInstances(1);
+            return UpdateControl.updateResource(kc);
```

Comment:
What if some other controller wants to manage the `instances` field via SSA? Do we really need to update the CRs spec from the Operator? Can't we simply set the default as:
```java
private Integer instances = 1;
```

Reply:
> What if some other controller wants to manage the instances field via SSA?

That's fine, it would end up as non-null.

> Do we really need to update the CRs spec from the Operator? Can't we simply set the default as:

Possibly.  There are two considerations:  
1. It's not consistent with built-in fabric8 object handling (or go I believe).  You have to know what fields to set to null to get the defaults.
2. On the server side we can't differentiate between 1 being explicitly set, or null.  However the field needs to explicitly be populated for the scale subresource calls to work, so we'd need at least need to do a little more - move the default to the getter and add another method to test whether it's actually null so that you can then trigger an update of the resource.

Reply:
A silly question, can’t we just rely on [defaulting](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting)?

Reply:
Unfortunately not easily -, https://github.com/fabric8io/kubernetes-client/issues/5324

Reply:
Ok, let's stick to approach in this PR and perhaps switch to K8s native defaulting once https://github.com/fabric8io/kubernetes-client/issues/5324 is resolved.